urllib3/connectionpool.py

```diff
@@ -666,6 +666,10 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                                 release_conn=release_conn, body_pos=body_pos,
                                 **response_kw)
 
+        def drain_and_release_conn(response):
+            response.read()
```

Comment:
Here's a question: should we catch exceptions here? I feel like the answer is probably yes: they'll be ultimately a bit misleading if we hit them.